File: day12/sort_algorithm/04_shell_sort.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        gap = n // 2
        while gap:
            for i in range(gap, n):
                while i - gap >= 0 and nums[i - gap] > nums[i]:
                    nums[i - gap], nums[i] = nums[i], nums[i - gap]
                    i -= gap
            gap //= 2
        return nums

File: day12/sort_algorithm/04_ShellSort.java
class Solution {
    public int[] sortArray(int[] nums) {
        int gap = nums.length / 2;
        while (gap > 0) {
            for (int i = gap; i < nums.length; i++) {
                while (i - gap >= 0 && nums[i - gap] > nums[i]) {
                    int tmp = nums[i - gap];
                    nums[i - gap] = nums[i];
                    nums[i] = tmp;
                    i--;
                }
            }
            gap /= 2;
        }
        return nums;
    }
}
